source/deployment/cluster.rst

```diff
@@ -0,0 +1,207 @@
+
+It's highly recommended that at least one read replica have a minimum of **three times the disk space** of the master database. This way, if the master database signals a "low disk space" warning the administrator can switch over to the read replica with a larger disk and continue operation, while also having room to locally duplicate and manipulate the production database. 
+
+
+Deploying a multi-database configuration 
+````````````````````````````````````````````````````````````````````````
+To configure a multi-database Mattermost server: 
+
+1. Update the ``DataSource`` setting ``config.json`` with a connection string to your master database server based on the database type set in ``DriverName``, which would be either ``postgres`` or ``mysql``. 
+2. Update the ``DataSourceReplicas`` setting ``config.json`` with a series of connection strings to your database read replica servers in the format ``["readreplia1", "readreplia2"]``. Each connection should also be compatible with the ``DriverName`` setting.
+
+The new settings can be applied by either stopping and starting the server, or by loading the configuration settings as described in the next section. 
+
+Once loaded, database write requests will be sent to the master database and read requests will be distributed among the other databases in the list.
+
+Loading a multi-database configuration onto an active server
+``````````````````````````````````````````````````````````````````````````````````````````````````
+After a multi-database configuration has been defined in ``config.json`` the following procedure can be used to apply the settings without shutting down the Mattermost server: 
+
+1. Go to **System Console** > **Configuration** and press **Reload Configuration from Disk** to reload configuration settings for the Mattermost server from ``config.json``. 
+2. Go to **System Console** > **Database** and press **Recycle Database Connections** to takedown existing database connections and set up new connections in the multi-database configuration. 
+
+While connection settings are changing there may be a brief moment when writes to the master database will be unsuccessful. The process waits for all existing connections to finish and starts serving new requests with the new connections. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
+
+Manual failover for master database  
+`````````````````````````````````````````````````````````````````
+If the need arises to switch from the current master database--for example, if it is running out of disk space, or requires maintenance updates, or for other reasons--the Mattermost server can switch to using one of its read replicas as a master database by updating `DataSource` in `config.json` the following procedure can be used to apply the settings without shutting down the Mattermost server: 
```

Comment:
Use double quotes around DataSource and config.json.

End with a period after `config.json.